src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
 public class FileWork {
+
+    private static final char FILTER_CHARACTER = 'w';
+
     public String[] readFromFile(String fileName) {
-        return null;
+        if (fileName == null) {
+            return new String[0];
+        }
+        File file = new File(fileName);
+        if (file.length() == 0) {
+            return new String[0];
+        }
+
+        String fileContent = getFileContent(file);
+        String[] allWords = fileContent.split("\\s+");
+        for (int i = 0; i < allWords.length; i++) {
+            allWords[i] = removeNonLetters(allWords[i]);
+        }
+
+        String[] sortedFilteredArray = getLinesStartingAt(FILTER_CHARACTER, allWords);
+        Arrays.sort(sortedFilteredArray);
+        return sortedFilteredArray;
+    }
+
+    private String[] getLinesStartingAt(char charToStartAt, String[] lines) {
```

Comment:
```suggestion
    private String[] getLinesStartingAt(String[] lines) {
```
it is redundant to pass charToStartAt here,  because we have a constant at this class